### PR TITLE
Selection script filtrage

### PR DESCRIPTION
**Pull request by Aminelouati**

### CODE REVIEW

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java [hunk 1]
```diff
@@ -263,10 +269,19 @@ private NodeSet doSelectNodes(Criteria criteria, Client client) {
...
             // arranging nodes for script execution
             List<RMNode> arrangedNodes = arrangeNodesForScriptExecution(afterPolicyNodes, criteria.getScripts());
-
             if (criteria.getTopology().isTopologyBased()) {
-                // run scripts on all available nodes
-                matchedNodes = runScripts(arrangedNodes, criteria);
+
+                List<RMNode> arrangedFilteredNodes = topologyNodesFilter.filterNodes(criteria, arrangedNodes);
+
+                if (arrangedFilteredNodes.size() == 0) {
```
Comment (marcocast) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java]: you can change this .size() == 0 with isEmpty()

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java [hunk 2]
```diff
@@ -263,10 +269,19 @@ private NodeSet doSelectNodes(Criteria criteria, Client client) {
...
+                    if (loggerIsDebugEnabled) {
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java]: this debug message is not necessary

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 8 issues

1. MAJOR [SelectionManager.java#L277](https://github.com/ow2-proactive/scheduling/blob/b7573524d4d0943687e674966b7140be63c83059/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java#L277): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)
1. MAJOR [TopologyNodesFilter.java#L48](https://github.com/ow2-proactive/scheduling/blob/b7573524d4d0943687e674966b7140be63c83059/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyNodesFilter.java#L48): Remove this unused "logger" private field. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1068)
1. MINOR [SelectionManager.java#L35](https://github.com/ow2-proactive/scheduling/blob/b7573524d4d0943687e674966b7140be63c83059/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java#L35): Remove this unused import 'java.util.LinkedHashMap'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [SelectionManager.java#L68](https://github.com/ow2-proactive/scheduling/blob/b7573524d4d0943687e674966b7140be63c83059/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/SelectionManager.java#L68): Remove this unused import 'org.ow2.proactive.topology.descriptor.TopologyDescriptor'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [TopologyNodesFilter.java#L28](https://github.com/ow2-proactive/scheduling/blob/b7573524d4d0943687e674966b7140be63c83059/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyNodesFilter.java#L28): Remove this unused import 'java.util.ArrayList'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [TopologyNodesFilter.java#L35](https://github.com/ow2-proactive/scheduling/blob/b7573524d4d0943687e674966b7140be63c83059/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyNodesFilter.java#L35): Remove this unused import 'org.ow2.proactive.resourcemanager.authentication.Client'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [TopologyNodesFilter.java#L37](https://github.com/ow2-proactive/scheduling/blob/b7573524d4d0943687e674966b7140be63c83059/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyNodesFilter.java#L37): Remove this unused import 'org.ow2.proactive.resourcemanager.selection.SelectionManager'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [TopologyNodesFilter.java#L48](https://github.com/ow2-proactive/scheduling/blob/b7573524d4d0943687e674966b7140be63c83059/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyNodesFilter.java#L48): Reorder the modifiers to comply with the Java Language Specification. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues

1. MAJOR [TopologyNodesFilter.java#L45](https://github.com/ow2-proactive/scheduling/blob/ffebbd7cedc68b108e9ac0d96066ff7858fd8de8/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyNodesFilter.java#L45): Remove this unused "logger" private field. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1068)
1. MINOR [TopologyNodesFilter.java#L45](https://github.com/ow2-proactive/scheduling/blob/ffebbd7cedc68b108e9ac0d96066ff7858fd8de8/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyNodesFilter.java#L45): Reorder the modifiers to comply with the Java Language Specification. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue

1. MINOR [TopologyNodesFilter.java#L33](https://github.com/ow2-proactive/scheduling/blob/09525f87d5cefb9534181b401e6efae15eeeb915/rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyNodesFilter.java#L33): Remove this unused import 'org.apache.log4j.Logger'. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by Aminelouati** [comment]
Retest this please

**Pull request merged**